core/src/test/scala/unit/kafka/admin/ReassignPartitionsCommandTest.scala

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package kafka.admin
+
+import java.io.{FileOutputStream, OutputStreamWriter, BufferedWriter, File}
+
+import kafka.admin.ReassignPartitionsCommand.ReassignPartitionsCommandOptions
+import kafka.utils.{TestUtils, ZkUtils, Logging}
+import kafka.zk.ZooKeeperTestHarness
+import org.junit.{After, Before, Test}
+import org.scalatest.junit.JUnit3Suite
+
+class ReassignPartitionsCommandTest extends ZooKeeperTestHarness with Logging with RackAwareTest {
+  var file: File = null;
+
+  @Before
+  override def setUp(): Unit = {
+    super.setUp()
+    file = createTmpFile()
+  }
+
+  @After
+  override def tearDown(): Unit = {
+    super.tearDown()
+    if (file != null) {
+      file.delete()
+    }
+  }
+
+  def createTmpFile(): File = {
+    val configFile = File.createTempFile("move", ".json");
+    val writer = new BufferedWriter(new OutputStreamWriter(new FileOutputStream(configFile), "UTF-8"))
+    val content = "{\"topics\": [{\"topic\": \"foo\"}], \"version\":1}"
+    writer.write(content)
+    writer.close()
+    configFile
+  }
+
+  @Test
+  def testRackAwareReassign(): Unit = {
```

Comment:
Do we need to start ZK in this test? It seems that we can just test AdminUtils.assignReplicasToBrokers() directly?